### PR TITLE
Imeplementation of Dynamic Meta Tags

### LINKED ISSUE #208: Implement Dynamic Meta Tags for Each Page
**What do we need to build or fix?**
Our Next.js application currently uses a static `Meta` component for SEO, which does not adapt to different pages. This limits our SEO effectiveness as each page should ideally have unique meta tags (title, description, keywords) and canonical URLs for better search engine optimization. This change will be an extension of the work we're already doing on #206.

**Technical details**
The objective is to enhance the `Meta` component to dynamically update meta tags based on the current page. This will involve integrating the Next.js router to identify the current route and retrieving corresponding meta details from a `meta.js` file in the `utils` directory.

**Approach suggestions**

- Modify the `meta.js` file to include a dictionary of routes with their respective meta information, including canonical URLs. For example:

```
// utils/meta.js
const pageMeta = {
  '/': {
    title: 'Home - Web Dev Path',
    description: 'Start your journey in web development with Web Dev Path.',
    keywords: 'web development, learning, coding',
    canonical: 'https://www.webdevpath.co',
  },
  '/about': {
    title: 'About Us - Web Dev Path',
    description: 'Learn about the Web Dev Path team and our mission.',
    keywords: 'about web dev path, web development team, web dev mentoring',
    canonical: 'https://www.webdevpath.co/about',
  },
  // Add other routes and their meta information here
};

export default pageMeta;
```

- Update the `Meta` component to use the Next.js router for fetching the relevant meta information based on the current route, including the canonical URL.

```
// components/layout/Meta.js
import { useRouter } from 'next/router';
import Head from 'next/head';
import pageMeta from '@/utils/meta';

export default function Meta() {
  const router = useRouter();
  const currentPath = router.asPath.split('?')[0]; // Get the path without query parameters
  const meta = pageMeta[currentPath] || pageMeta['/']; // Fallback to home page meta

  return (
    <Head>
      <title>{meta.title}</title>
      <meta name='description' content={meta.description} />
      <meta name='keywords' content={meta.keywords} />
      <link rel='canonical' href={meta.canonical} />
      {/* ... other meta tags ... */}
    </Head>
  );
}
```
- Remove the static canonical link from _document.js to avoid conflicts with the dynamic canonical URLs.
- Ensure the `Meta` component is correctly implemented in the Layout component, which wraps all pages in `_app.js`.
- Include a fallback mechanism in `Meta` for routes not explicitly defined in `meta.js`.
- _Collaboration Note:_ Please coordinate with @mariana-caldas  for the content of the meta.js properties for each page to ensure consistency and accuracy in our SEO efforts.

**Deadline**
Please keep in mind that once you assign this task to yourself, you'll need to complete it in 10 days.

**Acceptance criteria**

- Test the section and components in many screen sizes, you can use the Inspect tool for that.
- Please test if the new changes added to the components do not affect the other instances.
- Test the feature in many browsers, such as Chrome, Firefox, Edge, and Safari (MAC).
- If there are any build problems when submitting your PR, run `yarn build` locally to solve the issues and commit the changes.
- Update the CHANGELOG.md file.

### PR DESCRIPTION
**Pull request by Satoshi-Sh**
|                                    Web Dev Path                                    |
| :--------------------------------------------------------------------------------: |
| [**208**](https://github.com/MarianaSouza/web-dev-path/issues/208) |

closes #208

#### Have you updated the CHANGELOG.md file? If not, please do it.
Yes, I did.

#### Requirements 
- [x] Remove the static canonical link from _document.js to avoid conflicts with the dynamic canonical URLs.
- [x] Ensure the Meta component is correctly implemented in the Layout component, which wraps all pages in _app.js.
- [x] Include a fallback mechanism in Meta for routes not explicitly defined in meta.js.
- [x] Collaboration Note: Please coordinate with @mariana-caldas for the content of the meta.js properties for each page to ensure consistency and accuracy in our SEO efforts.

#### What is this change?

- Updated pageMeta on the utils/meta.js file to implement meta tags dynamically for each page. 
- Used usePathname to get a current path and used ternary expression to implement a fallback mechanism when a route is not defined on pageMeta.
 
#### Were there any complications while making this change?
No
#### How to replicate the issue?
N/A
#### If necessary, please describe how to test the new feature or fix it.
Run the app locally or use the preview page, and open the element tab to check the title, meta(keywords and description), and link(canonical) tags on your browser. They should match with the info in utils/meta.js file. 

To check the fallback mechanism, you can type a URL like  "localhost:3010/newpage".  Meta tags should be the same as "/".

#### When should this be merged?
After getting reviewed.

### CODE REVIEW

**Comment by Satoshi-Sh** [comment]
@mariana-caldas  Let me know if you want to change the pageMeta properties for the blog and contact.

File: components/layout/Meta.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import Head from 'next/head';
-import { meta } from '@/utils/meta';
+import pageMeta from '@/utils/meta';
+import { usePathname } from 'next/navigation';
 import Script from 'next/script';
 
 export default function Meta() {
+  const currentPath = usePathname();
```
Comment (mariana-caldas) [inline components/layout/Meta.js:7]: Clean approach with `usePathname`, @Satoshi-Sh !

**Review by mariana-caldas — APPROVED** [review]
Such a great job, @Satoshi-Sh !
Thanks for also adding the remaining `meta` content. 🚀

**Review by cherylli — APPROVED** [review]

**Comment by Satoshi-Sh** [comment]
Mariana fixed that, @LuSilvaDeveloper 

Thanks for reviewing!!

**Pull request merged**